lib/facility.rb

```diff
@@ -1,14 +1,58 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, 
+              :address, 
+              :phone, 
+              :services,
+              :collected_fees,
+              :registered_vehicles
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(info)
+    @name = info[:name]
+    @address = info[:address]
+    @phone = info[:phone]
     @services = []
+    @collected_fees = 0
+    @registered_vehicles = []
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle) 
+    if @services.include? 'Vehicle Registration'
+      @registered_vehicles << vehicle
+      collect_fee(vehicle)
+      vehicle.register
+      vehicle.get_plate
+    end
+  end
+
+  def collect_fee(vehicle)
+    if vehicle.electric_vehicle?
+      @collected_fees += 200
+    elsif vehicle.antique?
+      @collected_fees += 25
+    else 
+      @collected_fees += 100
+    end
+  end
+
+  def administer_written_test(registrant)
+    if @services.include? 'Written Test'
+      registrant.pass_written_test
+    end
+  end
```

Comment:
I think these three methods needed more conditions. Not only did you need to check if the @services included something specific, but we needed to also check that the Registrant was a certain age AND also had their permit. How would you add these additional conditions in?